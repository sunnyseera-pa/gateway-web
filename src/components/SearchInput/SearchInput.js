import PropTypes from 'prop-types';
import React from 'react';
import { omit } from '../../configs/propTypes';
import Icon from '../Icon';
import Input from '../Input';
import { PROP_TYPES_INPUT } from '../Input/Input.propTypes';

const SearchInput = ({ value, onReset, ...outerProps }) => {
	const inputRef = React.useRef(null);

	const handleReset = React.useCallback(
		e => {
			if (onReset) onReset(e, inputRef.current);
		},
		[inputRef.current]
	);

	return (
		<Input
			name='search'
			inputRef={inputRef}
			iconPrepend={<Icon name='search' fill='green700' size='lg' />}
			iconAppend={!!value && <Icon name='clear' color='purple500' size='lg' role='button' onClick={handleReset} />}
			value={value}
			{...outerProps}
		/>
	);
};

SearchInput.propTypes = {
	...omit(PROP_TYPES_INPUT, ['inputRef']),
	onReset: PropTypes.func,
};

SearchInput.defaultProps = {
	className: 'ui-SearchInput',
};

export default SearchInput;
