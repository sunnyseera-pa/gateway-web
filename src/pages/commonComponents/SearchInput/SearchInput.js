import PropTypes from 'prop-types';
import React from 'react';
import { omit } from '../../../configs/propTypes';
import Icon from '../Icon';
import Input from '../Input';
import { PROP_TYPES_INPUT } from '../Input/Input.propTypes';

const SearchInput = ({ value, onReset, ...outerProps }) => {
	const inputRef = React.useRef(null);

	const handleReset = React.useCallback(
		e => {
			if (onReset) onReset(inputRef.current, e);
		},
		[inputRef.current]
	);

	return (
		<Input
			inputRef={inputRef}
			iconPrepend={<Icon name='search' size='lg' />}
			iconAppend={!!value && <Icon name='clear' onClick={handleReset} size='lg' role='button' />}
			value={value}
			{...outerProps}
		/>
	);
};

SearchInput.propTypes = {
	...omit(PROP_TYPES_INPUT, ['inputRef']),
	onReset: PropTypes.func,
};

SearchInput.defaultProps = {
	className: 'ui-SearchInput',
};

export default SearchInput;
