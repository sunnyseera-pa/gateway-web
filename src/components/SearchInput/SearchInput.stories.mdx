import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';

import SearchInput from './SearchInput';

<Meta
	title='Core/SearchInput'
	component={SearchInput}
	parameters={{
		docs: {
			source: {
				type: 'code',
			},
		},
	}}
/>

<Canvas>
	<Story name='Basic'>
		{props => {
			const [value, setValue] = React.useState();
			const handleReset = React.useCallback(() => {
				setValue('');
			}, []);
			const handleChange = React.useCallback(({ target: { value } }) => {
				setValue(value);
			}, []);
			return <SearchInput placeholder='Search with collection' onReset={handleReset} onChange={handleChange} value={value} width='350px' />;
		}}
	</Story>
</Canvas>

<ArgsTable of={SearchInput} />
