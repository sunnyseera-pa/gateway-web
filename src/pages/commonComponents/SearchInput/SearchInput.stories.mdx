import { Meta, Story, Canvas } from '@storybook/addon-docs';

import SearchInput from './SearchInput';

<Meta title='SearchInput' component={SearchInput} />

<Canvas>
	<Story name='Basic'>
		{props => {
			const [value, setValue] = React.useState();
			const handleReset = React.useCallback(() => {
				setValue('');
			}, []);
			const handleChange = React.useCallback(({ target: { value } }) => {
				setValue(value);
			}, []);
			return <SearchInput placeholder='Search with collection' onReset={handleReset} onChange={handleChange} value={value} />;
		}}
	</Story>
</Canvas>
