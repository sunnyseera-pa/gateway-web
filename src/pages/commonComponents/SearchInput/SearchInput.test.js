import { render } from '@testing-library/react';
import React from 'react';
import SearchInput from '.';

jest.mock('../Icon', () => {
	return props => <button {...props}>Icon</button>;
});

const props = {
	value: '',
	onReset: jest.fn(),
};

let wrapper;

describe('Given the SearchInput component', () => {
	describe('When it is rendered', () => {
		beforeEach(() => {
			wrapper = render(<SearchInput {...props} />, {
				wrapper: Providers,
			});
		});

		it('Then matches the previous snapshot', async () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then does not have a reset icon', () => {
			expect(wrapper.container.querySelector('[role="button"]')).toBeFalsy();
		});

		describe('And it is reset', () => {
			beforeAll(() => {
				wrapper = render(<SearchInput {...props} value='collection' />, {
					wrapper: Providers,
				});

				const reset = wrapper.container.querySelector('[role="button"]');

				fireEvent.click(reset);
			});

			it('Then calls onReset', () => {
				expect(props.onReset).toHaveBeenCalled();
			});
		});
	});
});
