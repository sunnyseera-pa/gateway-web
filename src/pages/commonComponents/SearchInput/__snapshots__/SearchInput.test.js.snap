// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SearchInput component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="ui-Input ui-SearchInput css-jlq4v1 input-group"
  >
    <div
      class="css-rpufn0 input-group-prepend"
    >
      <button
        fill="purple500"
        name="search"
        role="button"
      >
        Icon
      </button>
    </div>
    <input
      class="form-control"
      value=""
    />
  </div>
</div>
`;
